### PR TITLE
Port from BN: Add command to pet menu for removing individual items from storage

### PR DESCRIPTION
**Pull request by WhiteCloud0123**
#### Summary

None

#### Purpose of change
Thie  is [Add command to pet menu for removing individual items from storage](https://github.com/cataclysmbnteam/Cataclysm-BN/pull/2868) in BN.I think this can further enhance the practicality of pets. It is also great for players who like pets, and transplantation does not require much work.

#### Describe the solution
Add a new function `take_items_from (Monster& z)` and new enumeration item in monexamine.cpp, and apply them in fact.
And I adjusted the position of 'Remove bag from' slightly, placing it later may be better.

#### Testing

After：
show_02
CLB_8{6 ONR1X284BD(OPH7
show_01

I generated a cow and then give and take items multiple times normally.

### CODE REVIEW

**Comment by andrei8l** [comment]
Please preserve commit authorship. You can add this text to the bottom of a commit to automatically credit the authors:

```

Co-authored-by: Chaosvolt <chaosvolt@users.noreply.github.com>
Co-authored-by: scarf005 <scarf005@users.noreply.github.com>
```
(though a cherry-pick would be better)

**Comment by WhiteCloud0123** [comment]
> Please preserve commit authorship. You can add this text to the bottom of a commit to automatically credit the authors:
> 
> ```
> 
> Co-authored-by: Chaosvolt <chaosvolt@users.noreply.github.com>
> Co-authored-by: scarf005 <scarf005@users.noreply.github.com>
> ```
> 
> (though a cherry-pick would be better)

Thanks.

**Comment by WhiteCloud0123** [comment]
I'll try to fill in the collaborators later.

File: src/monexamine.cpp [hunk 1]
```diff
@@ -281,6 +281,37 @@ bool give_items_to( monster &z )
...
+    uilist selection_menu;
+    selection_menu.text = string_format( _( "Select an item to remove from the %s." ), pet_name );
+    selection_menu.addentry( i++, true, MENU_AUTOASSIGN, _( "Cancel" ) );
+    for( const item &iter : monster_inv ) {
+        selection_menu.addentry( i++, true, MENU_AUTOASSIGN, _( "Retrieve %s" ), iter.tname() );
+    }
+    selection_menu.selected = 1;
+    selection_menu.query();
+    const int index = selection_menu.ret;
+    if( index == 0 || index == UILIST_CANCEL || index < 0 ||
```
Comment (Night-Pryanik) [inline src/monexamine.cpp:299]:
```suggestion
    if( index <= 0 || index == UILIST_CANCEL ||
```

**Comment by WhiteCloud0123** [comment]
In order to avoid polluting the submission records, I have decided to open another PR.

**Pull request closed without merging**